javascript/javascript3/week3/Shopping Cart/shoppingCart.js

```diff
@@ -0,0 +1,135 @@
+const reducer = (acc, currentVal) => acc + currentVal;
+const apiKey = "8df58438c8513b601e88d88777ea74e8";
+
+const arrayOfProducts = [];
+
+const listItem1 = document.getElementById("list-item-1");
+const listItem2 = document.getElementById("list-item-2");
+
+const userNamePlace = document.getElementById("user-name-spot");
+const pricePlace = document.getElementById("total-price");
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    //base currency is euro
+    const currencyConventerAPI = `http://data.fixer.io/api/latest?access_key=${apiKey}`;
+    fetch(currencyConventerAPI)
+      .then(res => res.json())
+      .then(data =>
+        console.log(
+          `${this.name} is ${Math.round(
+            this.price * data.rates[currency]
+          )} ${currency}`
+        )
+      );
+  }
+}
+
+const gum = new Product("gum", 2);
+gum.convertToCurrency("DKK");
+gum.convertToCurrency("USD");
+gum.convertToCurrency("ISK");
+
+class ShoppingCart {
+  constructor(products) {
+    this.products = products;
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    this.products.forEach(item => {
+      if (this.products.indexOf(product) !== -1) {
+        const filtered = this.products.splice(
+          this.products.indexOf(product),
+          1
+        );
+        return filtered;
+      }
+    });
+  }
+
+  searchProduct(productName) {
+    const product = [];
+    this.products.forEach(item => {
+      if (item.name.includes(productName)) {
+        return product.push(item);
+      }
+    });
+    console.log(product);
+    return product;
+  }
+
+  getTotal() {
+    const productPrices = [];
+    this.products.forEach(item => productPrices.push(item.price));
+    const sum = productPrices.reduce(reducer);
```

Comment:
it is confusing to have the reducer OUTSIDE the reduce, it is also needlessly complicated to use a reduce for a simple sum. But it's correct

Reply:
So using for loop is a better practice in this situation or something else?

Reply:
Looking around the internet, it seems like people like reduce. So it might be a matter of personal taste. However, i would still keep the "reducer" in the same place as the "reduce" because you don't have to search too far to understand what the code is doing.